src/server/session.ts

```diff
@@ -2885,14 +2900,21 @@ namespace ts.server {
         }
 
         public executeCommand(request: protocol.Request): HandlerResponse {
-            const handler = this.handlers.get(request.command);
-            if (handler) {
-                return this.executeWithRequestId(request.seq, () => handler(request));
+            const tracingData: tracing.EventData = [tracing.Phase.Session, "executeCommand", { seq: request.seq, command: request.command }];

```

Comment:
Again we should try to use existing try catch and do this from onMessage instead ?